src/core/thread/key_manager.cpp

```diff
@@ -443,4 +509,141 @@ void KeyManager::HandleKeyRotationTimer(void)
     }
 }
 
+void KeyManager::GetNetworkKey(NetworkKey &aNetworkKey) const
+{
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+    if (Crypto::Storage::IsKeyRefValid(mNetworkKeyRef))
+    {
+        Error  error = kErrorNone;
+        size_t keyLen;
+
+        error = Crypto::Storage::ExportKey(mNetworkKeyRef, aNetworkKey.m8, NetworkKey::kSize, keyLen);
+        OT_ASSERT(error == kErrorNone);
+        OT_ASSERT(keyLen == NetworkKey::kSize);
+        OT_UNUSED_VARIABLE(error);
+    }
+    else
+    {
+        aNetworkKey.Clear();
+    }
+#else
+    aNetworkKey = mNetworkKey;
+#endif
+}
+
+void KeyManager::GetPskc(Pskc &aPskc) const
+{
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+    if (Crypto::Storage::IsKeyRefValid(mPskcRef))
+    {
+        Error  error = kErrorNone;
+        size_t keyLen;
+
+        error = Crypto::Storage::ExportKey(mPskcRef, aPskc.m8, Pskc::kSize, keyLen);
+        OT_ASSERT(error == kErrorNone);
+        OT_ASSERT(keyLen == Pskc::kSize);
+        OT_UNUSED_VARIABLE(error);
+    }
+    else
+    {
+        aPskc.Clear();
+    }
+#else
+    aPskc = mPskc;
+#endif
+}
+
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+
+void KeyManager::StoreNetworkKey(const NetworkKey &aNetworkKey, bool aOverWriteExisting)
+{
+    Error         error;
+    NetworkKeyRef keyRef;
+
+    keyRef = kNetworkKeyPsaItsOffset;
+
+    if (!aOverWriteExisting)
+    {
+        Crypto::Storage::KeyAttributes keyAttributes;
+
+        error = Crypto::Storage::GetKeyAttributes(keyRef, keyAttributes);
+
+        // We will be able to retrieve the keyAttributes only if there is
+        // already a network key stored in ITS. If stored, and we are not
+        // overwriting the existing key, return without doing anything.
+        if (error == kErrorNone)
```

Comment:
@hemanth-silabs another idea/suggestion on `otPlatCryptoGetKeyAttributes()`?
- We only use it from one place in the code and from here.
- We are using to determine if the persistent key (with known `KeyRef`) value is present in the storage or not
- We currently don't use the `keyAttributes` bytes we get from calling this API (are there any use-cases for this in future)?

Can we consider using the  `otPlatCryptoExportKey()` for this purpose?
- Have it return `OT_ERROR_NOT_FOUND` if the given `KeyRef` cannot be find in the storage.

Or maybe replace `otPlatCryptoGetKeyAttributes()` with a simpler API that just checks the present of a `KeyRef` in storage? Something like:
```
// Indicate whether or not the PSA ITS contains a given `KeyRef`
bool otPlatCryptoHasKey(otCryptoKeyRef aKeyRef)
```

What do you think?

Reply:
I like the second idea better, as it does not involve exporting the key. We can rename the `otPlatCryptoGetKeyAttributes()` to `otPlatCryptoHasKey()` which returns a bool, but  internally still do the key attributes check. I say this because exportkey, might not work on non-exportable keys.

Reply:
@abtink I need to update the API_VERSION. I can push this change as well. What do you say?

Reply:
SGTM. Thanks.